chalice/app.py

```diff
@@ -230,11 +231,13 @@ class Request(object):
     """The current request from API gateway."""
 
     def __init__(self, query_params, headers, uri_params, method, body,
-                 context, stage_vars):
+                 context, stage_vars, is_base64_encoded):
         self.query_params = query_params
         self.headers = CaseInsensitiveMapping(headers)
         self.uri_params = uri_params
         self.method = method
+        if is_base64_encoded:
+            body = self._base64decode(body)
```

Comment:
We should be consistent with how json body is handled.  Any kind of processing on data should be done lazily.  We want the runtime to have as minimal overhead as possible.